src/CoreWCF.Primitives/src/CoreWCF/ServiceHostBase.cs

```diff
@@ -278,8 +278,12 @@ private ServiceAuthorizationBehavior EnsureAuthorization(ServiceDescription desc
 
             if (a == null)
             {
-                a = new ServiceAuthorizationBehavior();
-                description.Behaviors.Add(a);
+                // In ServiceHostBaseTests ServiceProvider is null..
+                a = description.ServiceProvider?.GetService(typeof(ServiceAuthorizationBehavior)) as ServiceAuthorizationBehavior;
+                if (a != null)
```

Comment:
@mconnew I would like but remove this null check but it is breaking ServiceHostBaseTests. I think `description.ServiceProvider` could not be null in real world. What do you think ?

Reply:
It doesn't harm having it there, but you're correct that it won't be null in the real world as we depend on so much from DI to start the service. We even pull IServiceBuilder from DI. Feel free to submit a follow up PR if you want to get rid of the null check, but I don't see harm in leaving it.